src/NuGet.Core/NuGet.Packaging/Signing/Verification/AllowListVerificationProvider.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Common;
+
+namespace NuGet.Packaging.Signing
+{
+    public class AllowListVerificationProvider : ISignatureVerificationProvider
+    {
+        private HashAlgorithmName _fingerprintAlgorithm;
+        private IEnumerable<VerificationAllowListObject> _allowList;
+        private bool _shouldCheckAllowList;
+
+        public AllowListVerificationProvider(HashAlgorithmName fingerprintAlgorithm, IEnumerable<VerificationAllowListObject> allowList)
+        {
+            _fingerprintAlgorithm = fingerprintAlgorithm;
+            _allowList = allowList;
+            _shouldCheckAllowList = _allowList != null && _allowList.Count() > 0;
```

Comment:
I suggest making `_allowList` an actual list to avoid multiple enumerations and the need to check it up front like this.